site/source/docs/porting/setjmp-longjmp.rst

```diff
@@ -0,0 +1,117 @@
+by using the buffer. ``longjmp``'s call stack should contain the function from
+which ``setjmp`` was called.
+
+Emscripten's support has a restriction that indirect calls to ``setjmp`` are not
+supported. For example, the below does not work:
+
+.. code-block:: c
+
+  jmp_buf env;
+  int (*fp)(jmp_buf) = setjmp;
+  fp(env); // Doesn't work
+
+
+JavaScript-based setjmp-longjmp Support
+=======================================
+
+In this mode, Emscripten emulates setjmp-longjmp using JavaScript. This option
+is set by adding ``-sSUPPORT_LONGJMP=emscripten`` to the command line, but this
+is currently enabled by default.
+
+Note that this option can have relatively high overhead in terms of code size,
+but it will work on all JavaScript engines with WebAssembly support, even if
+they do not yet support the `new WebAssembly exception handling proposal
+<https://github.com/WebAssembly/exception-handling/blob/master/proposals/exception-handling/Exceptions.md>`_.
+
+
+WebAssembly Exception Handling-based setjmp-longjmp Support
+===========================================================
+
+Alternatively, you can opt-in to the new support using the `WebAssembly
+exception handling
+<https://github.com/WebAssembly/exception-handling/blob/master/proposals/exception-handling/Exceptions.md>`_
+proposal. To enable it, pass ``-sSUPPORT_LONGJMP=wasm`` at both compile time and
+link time.
+
+This option leverages a new feature that brings built-in instructions for
+throwing and catching exceptions to WebAssembly. As a result, it can reduce code
+size and performance overhead compared to the JavaScript-based implementation.
+This option is currently supported in several major web browsers, but `may not
+be supported in all WebAssembly engines yet
+<https://webassembly.org/roadmap/>`_.
+
+
+.. _using-exceptions-and-setjmp-longjmp-together:
+
+Using Exceptions and setjmp-longjmp Together
+============================================
+
+We also have two kinds of :ref:`exception handling support <exceptions>`:
+JavaScript-based support and the new WebAssembly EH-based support. Our
+setjmp-longjmp support use the same mechanisms. Because of that, you should use
+the same kind of EH and setjmp-longjmp support when using exceptions and
+setjmp-longjmp together.
+
+For example, to use the JavaScript-based EH and setjmp-longjmp support together:
+
+.. code-block:: bash
+
+  em++ -fexceptions test.cpp -o test.js
+
+``-sSUPPORT_LONGJMP=1``, which defaults to ``emscripten``, is enabled by
+default, so you don't need to pass it explicitly.
+
+To use the WebAssembly EH and setjmp-longjmp support together:
+
+.. code-block:: bash
+
+  em++ -fwasm-exceptions -sSUPPORT_LONGJMP=wasm test.cpp -o test.js
+
+There is one restriction for using WebAssembly EH-based support for exceptions
+and setjmp-longjmp at the same time. You cannot call ``setjmp`` within a C++
+``catch`` clause. For example, the following will not work:
```

Comment:
Does this error at compile time or runtime? Might be worth mentioning.

Reply:
Compile time. Will specify.